### PR TITLE
store/tikv: fix GetRegion()/GetStore() issue caused by canceled context

### PR DESCRIPTION
**Pull request by hhkbp2**
Hi,

This PR fixes GetRegion()/GetStore() issue caused by cancelced context. It add the checking for GRPC `Canceled` error, so that a canceled context passed to GPRC interface would cause the region cache to be cleaned exceptionally.

PTAL @tiancaiamao @disksing @siddontang

### CODE REVIEW

File: store/tikv/region_request.go [hunk 1]
```diff
@@ -16,6 +16,9 @@ package tikv
 import (
 	"time"
 
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+
```
Comment (tiancaiamao) [inline store/tikv/region_request.go]: No need newline

File: store/tikv/region_request_test.go [hunk 2]
```diff
@@ -17,6 +17,9 @@ import (
 	"strings"
 	"time"
 
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+
```
Comment (tiancaiamao) [inline store/tikv/region_request_test.go]: ditto

File: store/tikv/region_request.go (hunk 1 above)
Comment (hhkbp2) [inline store/tikv/region_request.go]: Address comment.

**Comment by tiancaiamao** [comment]
LGTM

**Comment by disksing** [comment]
LGTM.

**Pull request merged**